directives/edit.js

```diff
@@ -157,7 +157,7 @@
 
           // WATCHERS
 
-          $scope.$watch('model.data.' + $scope.property, function (data) {
+          $scope.$watch('model.' + $scope.property, function (data) {
```

Comment:
I wouldn't hoist the watcher to the root model.  What do you need to watch up there?